Packages/com.unity.inputsystem/Tests/InputSystem/Monobehaviors/TrackedPoseDriverTests.cs

```diff
@@ -0,0 +1,155 @@
+using UnityEngine;
+using UnityEngine.TestTools;
+using NUnit.Framework;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine.SpatialTracking;
+using System.Linq;
+using UnityEngine.XR.PoseProvider;
+
+namespace UnityEngine.Experimental.Input.XR
+{
+    [TestFixture]
+    public class TrackedPoseDriverTests
+    {
+        public class TestTrackedPoseDriverWrapper : TrackedPoseDriver
+        {
+            public void FakeUpdate()
+            {
+                Update();
+            }
+
+            public void FakeOnBeforeRender()
+            {
+                OnBeforeRender();
+            }
+        }
+
+        static Vector3 testpos = new Vector3(1.0f, 2.0f, 3.0f);
+        static Quaternion testrot = new Quaternion(0.09853293f, 0.09853293f, 0.09853293f, 0.9853293f);
+
+
+        internal class TestPoseProvider : BasePoseProvider
+        {
+            public override bool TryGetPoseFromProvider(out Pose output)
+            {
+                Pose tmp = new Pose();                
+                tmp.position = testpos;
+                tmp.rotation = testrot;
+                output = tmp;
+                return true;
+            }
+        }
+
+        internal static TestTrackedPoseDriverWrapper CreateGameObjectWithTPD()
+        {
+            GameObject go = new GameObject();
+            TestTrackedPoseDriverWrapper tpd = go.AddComponent<TestTrackedPoseDriverWrapper>();
+            return tpd;
+        }
+
+        internal static BasePoseProvider CreatePoseProviderOnTPD(TestTrackedPoseDriverWrapper tpd)
+        {
+            TestPoseProvider tpp = tpd.gameObject.AddComponent<TestPoseProvider>();
+            tpd.poseProviderComponent = tpp;
+            return tpp;
+        }
+
+
+        [TearDown]
+        public void TearDown()
+        {
+
+        }
+
+        [Test]
+        public void TPDBindingTest()
+        {
+            var baseMap = new InputActionMap("Base");
+            var action1 = baseMap.AddAction("action1", binding: "<Gamepad>/buttonSouth");
+            var action2 = baseMap.AddAction("action2", binding: "<Gamepad>/buttonNorth");
+
+            TestTrackedPoseDriverWrapper tpd = CreateGameObjectWithTPD();
+            tpd.positionAction = action1;
+            Assert.That(tpd.positionAction.bindings[0].path, Is.EqualTo("<Gamepad>/buttonSouth"));
+            tpd.positionAction = action2;
+            Assert.That(tpd.positionAction.bindings[0].path, Is.EqualTo("<Gamepad>/buttonNorth"));
+
+            
+            tpd.rotationAction = action1;
+            Assert.That(tpd.rotationAction.bindings[0].path, Is.EqualTo("<Gamepad>/buttonSouth"));
+            tpd.rotationAction = action2;
+            Assert.That(tpd.rotationAction.bindings[0].path, Is.EqualTo("<Gamepad>/buttonNorth"));        
+        }

```

Comment:
This test doesn't test anything meaningful about TrackedPoseDriver. The checks are on functionality sitting in the core input system and covered by tests elsewhere.

There should be a test that triggers an action and then checks that TrackedPostDeriver has set the resulting transform as expected.

Reply:
I was trying to ensure that I correctly allowed changing the binding (as there's code in there which checks if the binding is bound and i want to make sure people dont screw that up). i think the better option is to do as tom suggests and check to ensure i unbind the option before i re-bind it.